tests/testthat/test-modspec.R

```diff
@@ -587,3 +587,64 @@ test_that("INPUT block", {
   expect_equal(tagdf$name, c("CL", "V2"))
   expect_equal(tagdf$tag, rep("input", 2))
 })
+
+test_that("Reserve names in cpp dot gh-1159", {
+  temp <- '$param {param}\n$main\n{main};\n$table\n{table}'
+  
+  param <- "cl = 1, bar = 2, vc = 5"
+  main <- "double b = 5;\nfoo.bar = true;"
+  table <- "true;"
+  code <- glue::glue(temp)
+  
+  expect_error(
+    mcode("cpp-dot-1", code, compile = FALSE), 
+    "Reserved symbol cannot be used as model name"
+  )
+  
+  table <- main
+  main <- "true;"
+  code <- glue::glue(temp)
+  
+  expect_error(
+    mcode("cpp-dot-2", code, compile = FALSE), 
+    "Reserved symbol cannot be used as model name"
+  )
+  
+  param <- "cl = 1, amt = 20"
+  main <- "double b = 5;\nself.amt = 100;"
+  table <- "true;"
+  code <- glue::glue(temp)
+  
+  expect_error(
+    mcode("cpp-dot-3", code, compile = FALSE), 
+    "invalid class"
+  )
+})
+
+test_that("Skip cpp dot check gh-1159", {
+  temp <- '$param {param}\n$main\n{main};\n$env MRGSOLVE_CPP_DOT_SKIP="foo"'
+  
+  param <- "cl = 1, foo = 3, vc = 5"
+  main <- "double b = 5;\nfoo.bar = true;"
+  table <- "true;"
+  code <- glue::glue(temp)
+  
+  expect_is(mcode("cpp-dot-skip-1", code, compile = FALSE), "mrgmod")
+  
+  temp <- '$param {param}\n$main\n{main};\n$env MRGSOLVE_CPP_DOT_SKIP="foo,bar"'
```

Comment:
I'm confused by this `MRGSOLVE_CPP_DOT_SKIP="foo,bar"`.  As I note above, `check_cpp_dot` receives these as a single string ("foo,bar") and doesn't have any logic to split them.  And if that were to work, I'd expect the `bar` check on line 640 to not signal an error.  

If the `$env MRGSOLVE_CPP_DOT_SKIP="foo,bar"` were dropped, the assertions would make sense to me.  Perhaps that's leftover from a previous iteration?  (In that case, that still leaves the question of whether comma-delimited values should be handled.)

Reply:
Thanks. Yeah, this doesn't make sense to me either now that look at it. I've refactored, including handling `MRGSOLVE_CPP_DOT_SKIP` properly.